src/github.com/matrix-org/go-neb/services/github/github.go

```diff
@@ -0,0 +1,99 @@
+package services
+
+import (
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/google/go-github/github"
+	"github.com/matrix-org/go-neb/database"
+	"github.com/matrix-org/go-neb/matrix"
+	"github.com/matrix-org/go-neb/plugin"
+	"golang.org/x/oauth2"
+	"regexp"
+	"strconv"
+)
+
+// Matches alphanumeric then a /, then more alphanumeric then a #, then a number.
+// E.g. owner/repo#11 (issue/PR numbers) - Captured groups for owner/repo/number
+const ownerRepoIssueRegex = "([A-z0-9-_]+)/([A-z0-9-_]+)#([0-9]+)"
+
+type githubService struct {
+	id     string
+	UserID string
+	Rooms  []string
+}
+
+func (s *githubService) ServiceUserID() string { return s.UserID }
+func (s *githubService) ServiceID() string     { return s.id }
+func (s *githubService) ServiceType() string   { return "github" }
+func (s *githubService) RoomIDs() []string     { return s.Rooms }
+func (s *githubService) Plugin(roomID string) plugin.Plugin {
+	return plugin.Plugin{
+		Commands: []plugin.Command{},
+		Expansions: []plugin.Expansion{
+			plugin.Expansion{
+				Regexp: regexp.MustCompile(ownerRepoIssueRegex),
```

Comment:
I'd prefer to have the regexp stored as a top-level variable, That way the MustCompile will fail earlier if the rexexp is bad. (Also means that we know that we will only construct the regexp once.)